xilem/src/view/async_repeat.rs

```diff
@@ -18,6 +29,13 @@ where
     H: Fn(&mut State, M) -> Action + 'static,
     M: Message + 'static,
 {
+    const {
+        assert!(
+            core::mem::size_of::<F>() == 0,
+            "`async_repeat` will not be ran again when its captured variables are updated.\n\
+            To ignore this warning, use `async_repeat_raw`."
```

Comment:
Yeah that's good to have, but there's no `async_repeat_raw`, and I don't think we should have it.
Same still applies to the `run_once_raw` IMO, though it doesn't hurt I guess, as it's not something the user reaches for first, and the user is warned with this assertion at least.